In [42]:
"""
Use the csv files created by the code, and put the final label for each text on each
lockdown, mask, and vaccination file for each dataset.
"""
import pandas as pd
import sklearn
from sklearn import metrics

# Primary Sets

# Dataset 0

In [193]:
df_0 = pd.read_csv('twitter_topic_0.csv')

In [194]:
df_0.fillna('missing', inplace=True)

In [195]:
df_0_lockdown = pd.read_csv('twitter_topic_0_lockdowns.csv')

In [196]:
df_0_mask = pd.read_csv('twitter_topic_0_masking_and_distancing.csv')

In [197]:
df_0_vaccination = pd.read_csv('twitter_topic_0_vaccination.csv')

In [198]:
#df_0.head()

# Dataset 1

In [199]:
df_1 = pd.read_csv('twitter_topic_1.csv')

In [200]:
df_1.fillna('missing', inplace=True)

In [201]:
df_1_lockdown = pd.read_csv('twitter_topic_1_lockdowns.csv')

In [202]:
df_1_mask = pd.read_csv('twitter_topic_1_masking_and_distancing.csv')

In [203]:
df_1_vaccination = pd.read_csv('twitter_topic_1_vaccination.csv')

In [204]:
#df_1.head()

# Dataset 2

In [205]:
df_2 = pd.read_csv('twitter_topic_2.csv')

In [206]:
df_2.fillna('missing', inplace=True)

In [207]:
df_2_lockdown = pd.read_csv('twitter_topic_2_lockdowns.csv')

In [208]:
df_2_mask = pd.read_csv('twitter_topic_2_masking_and_distancing.csv')

In [209]:
df_2_vaccination = pd.read_csv('twitter_topic_2_vaccination.csv')

In [210]:
#df_2.head()

# Dataset 3

In [211]:
df_3 = pd.read_csv('twitter_topic_3.csv')

In [212]:
df_3.fillna('missing', inplace=True)

In [213]:
df_3_lockdown = pd.read_csv('twitter_topic_3_lockdowns.csv')

In [214]:
df_3_mask = pd.read_csv('twitter_topic_3_masking_and_distancing.csv')

In [215]:
df_3_vaccination = pd.read_csv('twitter_topic_3_vaccination.csv')

In [216]:
#df_3.head()

# Dataset 4

In [217]:
df_4 = pd.read_csv('twitter_topic_4.csv')

In [218]:
df_4.fillna('missing', inplace=True)

In [219]:
df_4_lockdown = pd.read_csv('twitter_topic_4_lockdowns.csv')

In [220]:
df_4_mask = pd.read_csv('twitter_topic_4_masking_and_distancing.csv')

In [221]:
df_4_vaccination = pd.read_csv('twitter_topic_4_vaccination.csv')

In [222]:
#df_4.head()

## Finding Kappa

In [50]:
df_cur = df_2

kappa_1 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

kappa_2 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

kappa_3 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

kappa_4 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_0.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

In [51]:
print('kappa 1:', kappa_1)
print('kappa 2:', kappa_2)
print('kappa 3:', kappa_3)
print('kappa 4:', kappa_4)

kappa 1: 0.4125172568390408
kappa 2: 0.4479413860850894
kappa 3: 0.454406657189879
kappa 4: 0.2803217338357632


# Annotators, kappa's

## df_0: 
### 4 Annotators, keep all.

kappa 56: 0.37449258477361463

kappa 57: 0.4628485822445733

kappa 58: 0.36724433436200576

kappa 59: 0.48572536773342634


## df_1
### 4 Annotators, keep all.

kappa 91: 0.6210232245439785

kappa 85: 0.5957329504706256

kappa 86: 0.5975649954411598

kappa 87: 0.38390911860685667 


## df_2
### 4 Annotators, keep all.

kappa 51: 0.4724964451243631

kappa 52: 0.34995673694855817

kappa 53: 0.4956326545021735

kappa 54: 0.37251496342922535

## df_3
### 4 Annotators, keep all.

kappa 104: 0.44315735995110944

kappa 101: 0.65511568783928

kappa 102: 0.569458591203231

kappa 103: 0.3508589039050169

## df_4
### 4 Annotators, keep all.

kappa 72: 0.44315735995110944

kappa 74: 0.65511568783928

kappa 76: 0.569458591203231

kappa 71: 0.3508589039050169


# Labeling The Datasets
### Do it for each dataset

In [237]:
df_cur = df_4_lockdown
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 5] = True
    elif label_true < label_false:
        df_cur.iloc[i, 5] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 5] = True if kappa_scores>=0 else False
        
df_4_lockdown = df_cur

In [238]:
df_cur = df_4_mask
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 5] = True
    elif label_true < label_false:
        df_cur.iloc[i, 5] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 5] = True if kappa_scores>=0 else False
        
df_4_mask = df_cur

In [239]:
df_cur = df_4_vaccination
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 5] = True
    elif label_true < label_false:
        df_cur.iloc[i, 5] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 5] = True if kappa_scores>=0 else False
        
df_4_vaccination = df_cur

# Combine Datasets
### Combine each label type to itself

### Lockdown

In [242]:
df_lockdown = df_0_lockdown[['text', 'label']]
df_lockdown = df_lockdown.append(df_1_lockdown[['text','label']], ignore_index = True)
df_lockdown = df_lockdown.append(df_2_lockdown[['text','label']], ignore_index = True)
df_lockdown = df_lockdown.append(df_3_lockdown[['text','label']], ignore_index = True)
df_lockdown = df_lockdown.append(df_4_lockdown[['text','label']], ignore_index = True)

In [254]:
df_lockdown.head()

text  label
0     Federal Judge Rules Against CDC, Throws Out Cr...  False
1     Indeed. Even in the dysfunctional US health sy...  False
2     a vaccine seems to be ESSENTIAL if we are to s...  False
3     Nurses union calls on CDC to reinstate univers...  False
4     Sirf  #MukeshAmbani  Company &amp; it's Worker...  False
...                                                 ...    ...
1495  Dr Reddy's Laboratories and Russian Direct Inv...  False
1496  "#Covishield will be commercialised once the t...  False
1497  I am not, last lockdown here on RI was a compl...  False
1498   https://t.co/vuf52eSRgq  Why this warning or ...  False
1499  Opioid Prescribing Practices After the 2016 Re...  False

[1500 rows x 2 columns]

In [253]:
df_lockdown.to_csv("twitter_topic_lockdown.csv")

### Masking and Distancing

In [246]:
df_mask = df_0_mask[['text', 'label']]
df_mask = df_mask.append(df_1_mask[['text','label']], ignore_index = True)
df_mask = df_mask.append(df_2_mask[['text','label']], ignore_index = True)
df_mask = df_mask.append(df_3_mask[['text','label']], ignore_index = True)
df_mask = df_mask.append(df_4_mask[['text','label']], ignore_index = True)


In [255]:
df_mask.head()

text  label
0  Federal Judge Rules Against CDC, Throws Out Cr...  False
1  Indeed. Even in the dysfunctional US health sy...  False
2  a vaccine seems to be ESSENTIAL if we are to s...  False
3  Nurses union calls on CDC to reinstate univers...   True
4  Sirf  #MukeshAmbani  Company &amp; it's Worker...  False

In [256]:
df_mask.to_csv("twitter_topic_masking_and_distancing.csv")

### Vaccination

In [250]:
df_vaccination = df_0_vaccination[['text', 'label']]
df_vaccination = df_vaccination.append(df_1_vaccination[['text','label']], ignore_index = True)
df_vaccination = df_vaccination.append(df_2_vaccination[['text','label']], ignore_index = True)
df_vaccination = df_vaccination.append(df_3_vaccination[['text','label']], ignore_index = True)
df_vaccination = df_vaccination.append(df_4_vaccination[['text','label']], ignore_index = True)

In [258]:
df_vaccination.head()

text  label
0  Federal Judge Rules Against CDC, Throws Out Cr...  False
1  Indeed. Even in the dysfunctional US health sy...   True
2  a vaccine seems to be ESSENTIAL if we are to s...   True
3  Nurses union calls on CDC to reinstate univers...  False
4  Sirf  #MukeshAmbani  Company &amp; it's Worker...  False

In [259]:
df_vaccination.to_csv("twitter_topic_vaccination.csv")

# Secondary Sets
### Do the same procedure for kappa's and labeling

# Dataset 0

In [3]:
df_0 = pd.read_csv('changeorg_topic_0.csv')

In [4]:
df_0.fillna('missing', inplace=True)

In [5]:
df_0_lockdown = pd.read_csv('changeorg_topic_0_lockdowns.csv')

In [6]:
df_0_mask = pd.read_csv('changeorg_topic_0_masking_and_distancing.csv')

In [7]:
df_0_vaccination = pd.read_csv('changeorg_topic_0_vaccination.csv')

In [8]:
#df_0.head()

text annotation_43  \
0          Save Western Ghats from annihilation     lockdowns   
1                         Save Beacon Hill Park     lockdowns   
2  Government of India: Don't Tax Medical Bills     lockdowns   
3                              NO FAILING @ UCR     lockdowns   
4                WIAA - Let them play football!     lockdowns   

            annotation_44 annotation_45 annotation_46  
0                 missing       missing       missing  
1                 missing       missing       missing  
2                 missing       missing       missing  
3                 missing       missing       missing  
4  masking and distancing     lockdowns       missing

# Dataset 1

In [9]:
df_1 = pd.read_csv('changeorg_topic_1.csv')

In [10]:
df_1.fillna('missing', inplace=True)

In [11]:
df_1_lockdown = pd.read_csv('changeorg_topic_1_lockdowns.csv')

In [12]:
df_1_mask = pd.read_csv('changeorg_topic_1_masking_and_distancing.csv')

In [13]:
df_1_vaccination = pd.read_csv('changeorg_topic_1_vaccination.csv')

In [36]:
#df_1.head()

text annotation_73  \
0                                    COVID-19 Awards       missing   
1              Cancel the 2020-2021 school year SOLs     lockdowns   
2                                                 ..       missing   
3                                Protect our nurses!       missing   
4  SENSELESS IMPRISONMENT FOR MY MOM.. FIRST TIME...       missing   

  annotation_74 annotation_75 annotation_76  
0       missing       missing       missing  
1     lockdowns       missing     lockdowns  
2       missing       missing       missing  
3       missing       missing       missing  
4       missing       missing     lockdowns

# Dataset 2

In [15]:
df_2 = pd.read_csv('changeorg_topic_2.csv')

In [16]:
df_2.fillna('missing', inplace=True)

In [17]:
df_2_lockdown = pd.read_csv('changeorg_topic_2_lockdowns.csv')

In [18]:
df_2_mask = pd.read_csv('changeorg_topic_2_masking_and_distancing.csv')

In [19]:
df_2_vaccination = pd.read_csv('changeorg_topic_2_vaccination.csv')

In [37]:
#df_2.head()

text annotation_48  \
0                          Postpone CBSE Board Exams     lockdowns   
1                 Pay rise for paramedics and nurses       missing   
2                          Too Soon To Open Georgia!     lockdowns   
3  Request to reconsider DPS Ruby Park school tui...     lockdowns   
4  Covid-19 Aesthetics / Salon / Massage Industry...       missing   

  annotation_49 annotation_50 annotation_47  
0       missing       missing     lockdowns  
1       missing       missing       missing  
2     lockdowns     lockdowns     lockdowns  
3       missing       missing     lockdowns  
4       missing       missing     lockdowns

# Dataset 3

In [21]:
df_3 = pd.read_csv('changeorg_topic_3.csv')

In [22]:
df_3.fillna('missing', inplace=True)

In [23]:
df_3_lockdown = pd.read_csv('changeorg_topic_3_lockdowns.csv')

In [24]:
df_3_mask = pd.read_csv('changeorg_topic_3_masking_and_distancing.csv')

In [25]:
df_3_vaccination = pd.read_csv('changeorg_topic_3_vaccination.csv')

In [38]:
#df_3.head()

text annotation_112  \
0  Get CoVID Vaccines to Nepal ASAP and prevent a...    vaccination   
1                       Compensate Essential Workers        missing   
2  Amend Recent COVID-19 Restrictions for Persona...        missing   
3  Petition against COVID-19 & Quarantine Bill: S...      lockdowns   
4  KCL Has Coronavirus Case: Petition to Have Onl...      lockdowns   

  annotation_109 annotation_110                    annotation_111  
0    vaccination    vaccination                       vaccination  
1        missing    vaccination  masking and distancing,lockdowns  
2      lockdowns      lockdowns            masking and distancing  
3        missing        missing                           missing  
4      lockdowns      lockdowns                           missing

# Dataset 4

In [27]:
df_4 = pd.read_csv('changeorg_topic_4.csv')

In [28]:
df_4.fillna('missing', inplace=True)

In [29]:
df_4_lockdown = pd.read_csv('changeorg_topic_4_lockdowns.csv')

In [30]:
df_4_mask = pd.read_csv('changeorg_topic_4_masking_and_distancing.csv')

In [31]:
df_4_vaccination = pd.read_csv('changeorg_topic_4_vaccination.csv')

In [39]:
#df_4.head()

text annotation_96  \
0         Language Education in the Time of COVID-19       missing   
1                                 COVID-19 Test Kits       missing   
2                                 COVID 19 IN PRISON       missing   
3                                     Get Waled Home       missing   
4  Make pass/fail available for Mississippi State...       missing   

  annotation_53 annotation_93 annotation_94 annotation_95  
0       missing       missing       missing       missing  
1       missing       missing       missing       missing  
2       missing       missing       missing       missing  
3       missing       missing       missing       missing  
4       missing       missing       missing       missing

### Dataset 4 has five annotators, change some of the code while looking at the kappa

## Finding Kappa

In [54]:
df_cur = df_2

kappa_1 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

kappa_2 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

kappa_3 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

kappa_4 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_0.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/3

In [57]:
#optimized for dataset 4 (has 5 annotators)

df_cur = df_4

kappa_1 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 1], df_cur.iloc[:, 5], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/4

kappa_2 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 2], df_cur.iloc[:, 5], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/4

kappa_3 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 3], df_cur.iloc[:, 5], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/4

kappa_4 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_0.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 4], df_cur.iloc[:, 5], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/4

kappa_5 = (sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 5], df_cur.iloc[:, 1], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          + sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 5], df_0.iloc[:, 2], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 5], df_cur.iloc[:, 3], 
                                             labels=None, weights=None, 
                                             sample_weight=None)
          +sklearn.metrics.cohen_kappa_score(df_cur.iloc[:, 5], df_cur.iloc[:, 4], 
                                             labels=None, weights=None, 
                                             sample_weight=None))/4


In [58]:
print('kappa 1:', kappa_1)
print('kappa 2:', kappa_2)
print('kappa 3:', kappa_3)
print('kappa 4:', kappa_4)
print('kappa 4:', kappa_5)

kappa 1: 0.3056438218179325
kappa 2: 0.0479864857235143
kappa 3: 0.230724059332207
kappa 4: 0.315059519844437
kappa 4: 0.3070102812116629


# Annotators, kappa's

## df_0: 
### 3 Annotators. Annotator 43 is unreliable.

kappa 43: 0.0729400722183449

kappa 44: 0.3653958972415916

kappa 45: 0.3950997775171102

kappa 46: 0.38086750540434416


## df_1
### 4 Annotators, keep all.

kappa 73: 0.5998101568552087

kappa 74: 0.6064004363143914

kappa 75: 0.4588812351405374

kappa 76: 0.4014548691920357


## df_2
### 4 Annotators, keep all.

kappa 48: 0.4125172568390408

kappa 49: 0.4479413860850894

kappa 50: 0.454406657189879

kappa 47: 0.2803217338357632

## df_3
### 3 Annotators Annotator 111 is unreliable.

kappa 112: 0.3744886183380893

kappa 109: 0.41148335806861763

kappa 110: 0.32150079175224594

kappa 111: 0.17381735266047096

## df_4
### 4 Annotators. Annotator 53 is unreliable.

kappa 96: 0.3056438218179325

kappa 53: 0.0479864857235143

kappa 93: 0.230724059332207

kappa 94: 0.315059519844437

kappa 95: 0.3070102812116629


## Drop unreliable annotators

In [64]:
df_0 = df_0.drop(columns=['annotation_43'])
df_0_lockdown = df_0_lockdown.drop(columns=['annotation_43'])
df_0_mask = df_0_mask.drop(columns=['annotation_43'])
df_0_vaccination = df_0_vaccination.drop(columns=['annotation_43'])

In [118]:
#df_0_vaccination.head()

text  annotation_44  annotation_45  \
0          Save Western Ghats from annihilation          False          False   
1                         Save Beacon Hill Park          False          False   
2  Government of India: Don't Tax Medical Bills          False          False   
3                              NO FAILING @ UCR          False          False   
4                WIAA - Let them play football!          False          False   

   annotation_46  label  
0          False  False  
1          False  False  
2          False  False  
3          False  False  
4          False  False

In [66]:
df_3 = df_3.drop(columns=['annotation_111'])
df_3_lockdown = df_3_lockdown.drop(columns=['annotation_111'])
df_3_mask = df_3_mask.drop(columns=['annotation_111'])
df_3_vaccination = df_3_vaccination.drop(columns=['annotation_111'])

In [117]:
#df_3.head()

text  annotation_112  \
0  Get CoVID Vaccines to Nepal ASAP and prevent a...           False   
1                       Compensate Essential Workers           False   
2  Amend Recent COVID-19 Restrictions for Persona...           False   
3  Petition against COVID-19 & Quarantine Bill: S...            True   
4  KCL Has Coronavirus Case: Petition to Have Onl...            True   

   annotation_109  annotation_110  label  
0           False           False  False  
1           False           False  False  
2            True            True  False  
3           False           False  False  
4            True            True   True

In [114]:
df_4 = df_4.drop(columns=['annotation_53'])
df_4_lockdown = df_4_lockdown.drop(columns=['annotation_53'])
df_4_mask = df_4_mask.drop(columns=['annotation_53'])
df_4_vaccination = df_4_vaccination.drop(columns=['annotation_53'])

In [115]:
#df_4_lockdown

text  annotation_96  \
0           Language Education in the Time of COVID-19          False   
1                                   COVID-19 Test Kits          False   
2                                   COVID 19 IN PRISON          False   
3                                       Get Waled Home          False   
4    Make pass/fail available for Mississippi State...          False   
..                                                 ...            ...   
295  Cancellation of Lebanese Official Exams for 20...          False   
296  Hazard pay for essential employees working wit...          False   
297  Stop children being weighed in school without ...          False   
298               Allow Visitors for Covid-19 Patients           True   
299  Exigimos acciones de control y salud en la Res...          False   

     annotation_93  annotation_94  annotation_95 label  
0            False          False          False  None  
1            False          False          False  None  
2            False          False          False  None  
3            False          False          False  None  
4            False          False          False  None  
..             ...            ...            ...   ...  
295           True          False          False  None  
296          False          False          False  None  
297          False          False          False  None  
298          False          False          False  None  
299          False          False          False  None  

[300 rows x 6 columns]

# Labeling The Datasets
### Do it for each dataset

### For 4 annotations

In [119]:
df_cur = df_4_lockdown
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 5] = True
    elif label_true < label_false:
        df_cur.iloc[i, 5] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 5] = True if kappa_scores>=0 else False
        
df_4_lockdown = df_cur

In [120]:
#df_4_lockdown

text  annotation_96  \
0           Language Education in the Time of COVID-19          False   
1                                   COVID-19 Test Kits          False   
2                                   COVID 19 IN PRISON          False   
3                                       Get Waled Home          False   
4    Make pass/fail available for Mississippi State...          False   
..                                                 ...            ...   
295  Cancellation of Lebanese Official Exams for 20...          False   
296  Hazard pay for essential employees working wit...          False   
297  Stop children being weighed in school without ...          False   
298               Allow Visitors for Covid-19 Patients           True   
299  Exigimos acciones de control y salud en la Res...          False   

     annotation_93  annotation_94  annotation_95  label  
0            False          False          False  False  
1            False          False          False  False  
2            False          False          False  False  
3            False          False          False  False  
4            False          False          False  False  
..             ...            ...            ...    ...  
295           True          False          False  False  
296          False          False          False  False  
297          False          False          False  False  
298          False          False          False  False  
299          False          False          False  False  

[300 rows x 6 columns]

In [121]:
df_cur = df_4_mask
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 5] = True
    elif label_true < label_false:
        df_cur.iloc[i, 5] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 5] = True if kappa_scores>=0 else False
        
df_4_mask = df_cur

In [122]:
df_cur = df_4_vaccination
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 5] = True
    elif label_true < label_false:
        df_cur.iloc[i, 5] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 5] = True if kappa_scores>=0 else False
        
df_4_vaccination = df_cur

### For 3 annotations

In [98]:
df_cur = df_3_lockdown
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 4] = True
    elif label_true < label_false:
        df_cur.iloc[i, 4] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 4] = True if kappa_scores>=0 else False
        
df_3_lockdown = df_cur

In [99]:
df_cur = df_3_mask
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 4] = True
    elif label_true < label_false:
        df_cur.iloc[i, 4] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 4] = True if kappa_scores>=0 else False
        
df_3_mask = df_cur

In [100]:
df_cur = df_3_vaccination
df_cur['label'] = None

for i in range(len(df_cur)):
    label_true = 0
    label_false = 0
    
    for col in df_cur.loc[i][1:]:
        if col:
            label_true+=1
        else:
            label_false+=1
    
    if label_true > label_false:
        df_cur.iloc[i, 4] = True
    elif label_true < label_false:
        df_cur.iloc[i, 4] = False
    else: #labels are same
        kappa_scores = 0
        kappa_scores += kappa_1 if df_cur.loc[i][1] else -kappa_1
        kappa_scores += kappa_2 if df_cur.loc[i][1] else -kappa_2
        kappa_scores += kappa_3 if df_cur.loc[i][1] else -kappa_3
        kappa_scores += kappa_4 if df_cur.loc[i][1] else -kappa_4

        df_cur.iloc[i, 4] = True if kappa_scores>=0 else False
        
df_3_vaccination = df_cur

# Combine Datasets
### Combine each label type to itself

### Lockdown

In [123]:
df_lockdown_secondary = df_0_lockdown[['text', 'label']]
df_lockdown_secondary = df_lockdown_secondary.append(df_1_lockdown[['text','label']], ignore_index = True)
df_lockdown_secondary = df_lockdown_secondary.append(df_2_lockdown[['text','label']], ignore_index = True)
df_lockdown_secondary = df_lockdown_secondary.append(df_3_lockdown[['text','label']], ignore_index = True)
df_lockdown_secondary = df_lockdown_secondary.append(df_4_lockdown[['text','label']], ignore_index = True)

In [125]:
df_lockdown_secondary.head(-20)

text  label
0                  Save Western Ghats from annihilation  False
1                                 Save Beacon Hill Park  False
2          Government of India: Don't Tax Medical Bills  False
3                                      NO FAILING @ UCR  False
4                        WIAA - Let them play football!  False
...                                                 ...    ...
1475  Allow parents to temporarily home educate with...  False
1476  There will be a day AFTER COVID-19.  Help us c...  False
1477         Close Madison City Schools due to COVID-19   True
1478  Mthuli Ncube to remove 2% charge from all elec...  False
1479  Partially Refund Loughborough University Tuiti...  False

[1480 rows x 2 columns]

In [126]:
df_lockdown_secondary.to_csv("changeorg_topic_lockdown.csv")

### Masking and Distancing

In [127]:
df_mask_secondary = df_0_mask[['text', 'label']]
df_mask_secondary = df_mask_secondary.append(df_1_mask[['text','label']], ignore_index = True)
df_mask_secondary = df_mask_secondary.append(df_2_mask[['text','label']], ignore_index = True)
df_mask_secondary = df_mask_secondary.append(df_3_mask[['text','label']], ignore_index = True)
df_mask_secondary = df_mask_secondary.append(df_4_mask[['text','label']], ignore_index = True)

In [128]:
df_mask_secondary.head()

text  label
0          Save Western Ghats from annihilation  False
1                         Save Beacon Hill Park  False
2  Government of India: Don't Tax Medical Bills  False
3                              NO FAILING @ UCR  False
4                WIAA - Let them play football!  False

In [129]:
df_mask_secondary.to_csv("changeorg_topic_masking_and_distancing.csv")

### Vaccination

In [130]:
df_vaccination_secondary = df_0_vaccination[['text', 'label']]
df_vaccination_secondary = df_vaccination_secondary.append(df_1_vaccination[['text','label']], ignore_index = True)
df_vaccination_secondary = df_vaccination_secondary.append(df_2_vaccination[['text','label']], ignore_index = True)
df_vaccination_secondary = df_vaccination_secondary.append(df_3_vaccination[['text','label']], ignore_index = True)
df_vaccination_secondary = df_vaccination_secondary.append(df_4_vaccination[['text','label']], ignore_index = True)

In [131]:
df_vaccination_secondary.head()

text  label
0          Save Western Ghats from annihilation  False
1                         Save Beacon Hill Park  False
2  Government of India: Don't Tax Medical Bills  False
3                              NO FAILING @ UCR  False
4                WIAA - Let them play football!  False

In [132]:
df_vaccination_secondary.to_csv("changeorg_topic_vaccination.csv")